concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -247,6 +247,21 @@ pub struct consensus_runner {
     private: [u8; 0],
 }
 
+/// An opaque reference to an 'executable block' i.e., a block that
+/// where the metadata has been verified.
+/// The value behind the reference i.e. the "execute block continuation"
+/// is created in the consensus module but its owned here on the rust side.
+/// The value is being freed by the consensus side via 'executeBlock'.
+#[repr(C)]
+pub struct execute_block {
+    private: [u8; 0],
+}
+
+/// Abstracts the reference required to execute a block that has been received.
+pub struct ExecuteBlockCallback {
+    callback: *mut execute_block,
+}
```

Comment:
I neither understand the comment, nor the purpose of this type. You already have an opaque reference, the `execute_block`. Now you add another wrapper around it. This callback is also different than other things called "callback" below, which are all functions (type aliases for). This is a pointer which generally don't have such wrappers in this module.

Reply:
At the very least this should be `repr(transparent)`.

Reply:
The point is to prevent the the callback being run more than once, since it disposes of the reference.

Reply:
I've added repr(transparent). And @td202  is right about the underlying motivation for having this extra abstraction on top of the raw ptr.